weather.py

```diff
@@ -1,64 +1,100 @@
-                            mn_dt = line[0]
-                        if int(line[7]) >= mx_hum:
-                            mx_hum = int(line[7]) 
-                            hum_dt = line[0]
-
-
-    def convert_list_to_date(dt, seperator=''):
-        seperator.join(dt)
-        dt_obj = datetime.strptime(dt,'%Y-%m-%d')
-        c_dt = dt_obj.strftime("%b %d")
-        return seperator.join(c_dt)
-
-    # converted_dt = convert_list_to_string(mx_dt)
-    # dt_obj = datetime.strptime(converted_dt,'%Y-%m-%d')
-    # c_dt = dt_obj.strftime("%b %d")
-    cmx_dt = convert_list_to_date(mx_dt)
-    cmn_dt = convert_list_to_date(mn_dt)
-    chum_dt = convert_list_to_date(hum_dt)
-
-    print(f"Highest: {mx_tmp}C on {cmx_dt}")
-    print(f"Lowest: {mn_tmp}C on {cmn_dt}")
-    print(f"Humidity: {mx_hum}% on {chum_dt}")
-
-if __name__ == '__main__':
-    main()
-
-
+import os
+import sys
+import csv
+from datetime import datetime
+from pathlib import Path
+
+DATA_FOLDER = os.path.dirname(os.path.abspath(__file__))
+
+def main():
+    if str.upper(sys.argv[1]) == 'YEAR':
+        year = sys.argv[2]
+        year_data = year_calculator(year)
+        yearly_report(year_data)
+    elif str.upper(sys.argv[1]) == 'MONTH':
+        month = sys.argv[2].split('/')
+        month_data = month_calculator(month)
+        monthly_report(month_data)
+    else:
+        print("Please check input parameters")
+        exit
+
+'''Calculation based on year input'''
+
+def year_calculator(year):
+    max_tmp = 0
+    max_hum = 0
+    min_tmp = 100
+
+    with os.scandir(DATA_FOLDER) as data_files:
+        for file in data_files:
+            if year in file.name:
+                with open (f'{file.name}','r') as data_f:
+                        data_f.readline()    
+                        reader = csv.DictReader(data_f, restkey=None, restval=None)
+                        for line in reader:
+                            if line["Max TemperatureC"] is None or line["Max TemperatureC"] == '':
+                                continue
+                            else:
+                                if int(line["Max TemperatureC"]) >= max_tmp:
+                                    max_tmp = int(line["Max TemperatureC"])
+                                    maxtmp_dt = line["PKT"]
+                                if int(line["Min TemperatureC"]) <= min_tmp:
+                                    min_tmp = int(line["Min TemperatureC"]) 
+                                    mintmp_dt = line["PKT"]
+                                if int(line["Max Humidity"]) >= max_hum:
+                                    max_hum = int(line["Max Humidity"]) 
+                                    maxhum_dt = line["PKT"]
+            else:
+                pass
+    c_maxtmp_dt = string_to_date(maxtmp_dt)
+    c_mintmp_dt = string_to_date(mintmp_dt)
+    c_maxhum_dt = string_to_date(maxhum_dt)
+
+    return [max_tmp,c_maxtmp_dt,min_tmp,c_mintmp_dt,max_hum,c_maxhum_dt]
```

Comment:
This is okay but not good. Now I have to go back and forth in `yearly_report` function to see what are you passing in the arguments.

Reply:
Should try with some list or dict? or do you mean something else

Reply:
I think you must not return these in an array.